### ISSUE TITLE
Compiler message file broken: NullPointerException

### ISSUE THREAD

**Issue by lcardito** [corretto/corretto-11#299]
### Describe the bug
`java.lang.NullPointerException` during compilation

### To Reproduce
https://github.com/lcardito/amazoncorretto-11-compile

### Expected behavior
Code compiles as per previous versions

### Screenshots
_NA_

### Platform information
```
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: /opt/homebrew/Cellar/maven/3.8.6/libexec
Java version: 11.0.17, vendor: Amazon.com Inc., runtime: /Library/Java/JavaVirtualMachines/amazon-corretto-11.jdk/Contents/Home
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "12.2.1", arch: "aarch64", family: "mac"
```

### Additional context
None

**Comment by eastig** [comment]
Hi @lcardito,

Thank you for reporting the issue. I managed to reproduce the issue.
I found two similar reported issues:
https://bugs.openjdk.org/browse/JDK-8297557
https://bugs.openjdk.org/browse/JDK-8297336

**Comment by eastig** [comment]
It is reproducible with jdk-11.0.17+8: https://github.com/openjdk/jdk11u-dev/releases/tag/jdk-11.0.17%2B8 
It is not reproducible with jdk-11.0.18+1: https://github.com/openjdk/jdk11u-dev/releases/tag/jdk-11.0.18%2B1

**Comment by eastig** [comment]
As Corretto-11 synchronizes with OpenJDK11, I checked with Corretto-11 tip. The is not reproducible.
So the issue should disappear in its next release: 11.0.18.

**Comment by Gleb-Gadyatskiy** [comment]
It can also be https://bugs.openjdk.org/browse/JDK-8212586

**Comment by olivergillespie** [comment]
The fix was released in 11.0.18. Feel free to reopen if you still see the issue.
